src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -125,10 +125,10 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       case None =>
         val augmentedClusterRequest = addClusterDefaultLabels(serviceAccountInfo, googleProject, clusterName, userEmail, clusterRequest)
         val clusterFuture = for {
-          // Notify the auth provider that the cluster has been created
-          _ <- authProvider.notifyClusterCreated(userEmail, googleProject, clusterName)
           // Create the cluster in Google
           (cluster, initBucket, serviceAccountKeyOpt) <- createGoogleCluster(userEmail, serviceAccountInfo, googleProject, clusterName, augmentedClusterRequest)
+          // Notify the auth provider that the cluster has been created
+          _ <- authProvider.notifyClusterCreated(cluster)
```

Comment:
aww, crap. so this way around, if the cluster creation fails, the auth provider gets a delete that it never saw a corresponding create for. i really should have left a comment here -- this is on me. but this has got to stay where it is.

i spent some time wondering whether to make a pretend `Cluster` that has everything-but-the-googly-bits, but at this point it might be simpler to have a `ClusterAuthInfo` case class that just contains the relevant set of fields, which could be buildable from both a `ClusterRequest` and a `Cluster`.

let's discuss tomorrow, we can knock something out quickly if we work on it together